portfolio/src/main/java/com/google/sps/data/CuisineDatastore.java

```diff
@@ -0,0 +1,84 @@
+package com.google.sps.data;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.EnumMap;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import com.google.appengine.api.datastore.*;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.AvailableCuisines;
+
+/** Class that handles writing/reading cuisine data from the datastore. */
+public final class CuisineDatastore {
+    
+  private static final Logger log = Logger.getLogger(CuisineDatastore.class.getName());
+  public static final String CUISINE_ENTITY = "CuisineVotes";
+  public static final String CUISINE_PARAMETER = "cuisine";
+  public static final String USERID_PARAMETER = "userId";
+
+  private CuisineDatastore() {}
+
+    /** Method that retreives cuisine entities from the datastore and formats it. */
+    public static Map<AvailableCuisines, Integer> fetchCuisineVotes() {
+      Query query = new Query(CUISINE_ENTITY);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+      
+      EnumMap<AvailableCuisines, Integer> cuisineVotes = new EnumMap<>(AvailableCuisines.class);
+
+      for (AvailableCuisines cuisineId : AvailableCuisines.values()) {
+        cuisineVotes.put(cuisineId, 0);
+      }
+
+      for (Entity entity : results.asIterable()) {
+        try {
+          String dbCuisineId = (String) entity.getProperty(CUISINE_PARAMETER); //store the name of the enum (capitalized)
+          AvailableCuisines cuisineId = AvailableCuisines.getFromId(dbCuisineId);
+          Integer votes = cuisineVotes.get(cuisineId);
+          if (votes == null) {
+            votes = 0;
+          }
+          cuisineVotes.put(cuisineId, votes + 1);
+        } catch (IllegalArgumentException e) {
+          log.log(Level.SEVERE, "IllegalArg caught in fetchCuisineVotes", e);
```

Comment:
nit error message should be more like `"Cuisine not found for id: %s",cuisineId` rather than "illegal arg caught" which is exactly what the stacktrace will show anyway. You should move your `String dbCuisineId = getProperty()` outside of the try/catch